gstd/gstd_list.c

```diff
@@ -124,14 +128,25 @@ gstd_list_dispose (GObject * object)
 
   GST_INFO_OBJECT (self, "Disposing %s list", GSTD_OBJECT_NAME (self));
 
+  g_mutex_lock (&self->mutex);
```

Comment:
Objects in gstd are GstObjects, which already have a mutex we can use. Can we achieve the same effect by using `GST_OBJECT_LOCK(self)` and `GST_OBJECT_UNLOCK(self)`?

Reply:
I replaced the mutex lock as bellow.
https://github.com/RidgeRun/gstd-1.x/pull/225/commits/18ea54cd01ed4e932ba022bc7ab6c5fb93620aa8